### PR TITLE
add pylint github action and fix any resulting issues

### PR DESCRIPTION
**Pull request by ohnorobo**
Some of the lint compliance changes are improvements, some are just neutral, and a lot are just adding extra docstrings.
But better to get everything in compliance so this can run going forward.

pylint has bad default settings, so I've disabled or changed several things in setup.cfg

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -19,13 +19,18 @@ jobs:
         pip install -r requirements.txt
```
Comment (fortuna) [inline .github/workflows/build.yml:19]: Let's merge all the pip install calls.

File: table/run_queries.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def rebuild_all_tables() -> None:
   for filepath in glob.glob('table/queries/*.sql'):
     try:
       run_query(filepath)
-    except Exception as ex:
+    except Exception as ex:  # pylint: disable=broad-except
```
Comment (fortuna) [inline table/run_queries.py:39]: You may want to re-raise the exception instead of adding this pylint exception.

**Review by fortuna — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (ohnorobo) [reply .github/workflows/build.yml]: done

File: table/run_queries.py (hunk 2 above)
Comment (ohnorobo) [reply table/run_queries.py]: done

**Pull request merged**